addons/binding/org.openhab.binding.ikeatradfri/src/main/java/org/openhab/binding/ikeatradfri/handler/IkeaTradfriBulbHandler.java

```diff
@@ -152,41 +146,31 @@ public void onDataUpdate(JsonElement data) {
     }
 
     private void updateStatusFromProperties(LightProperties props) {
-        try {
-            OnOffType onoff = props.getOnOffState() ? OnOffType.ON : OnOffType.OFF;
-            updateState(CHANNEL_STATE, onoff);
-            logger.debug("Updating channel state: {}", onoff.toString());
+        if (!props.getOnOffState()) {
+            updateState(CHANNEL_BRIGHTNESS, PercentType.ZERO);
+            logger.debug("Updating channel brightness: {}", 0);
         }
-        catch(NullPointerException e) {}
 
-        try {
-            PercentType dimmer = props.getBrightness();
+        PercentType dimmer = props.getBrightness();
+        if (dimmer != null) {
             updateState(CHANNEL_BRIGHTNESS, dimmer);
             logger.debug("Updating channel brightness: {} from {}", dimmer.toString(), props.getBrightness());
         }
-        catch(NullPointerException e) {}
 
-        try {
-            PercentType colorTemp = props.getColorTemperature();
+        PercentType colorTemp = props.getColorTemperature();
```

Comment:
Ok, thanks for the hint! Imho it is pretty unexpected that a `get...` method results in an NPE, so I rather have fixed the implementation of the method with https://github.com/openhab/openhab2-addons/pull/2200/commits/b41085a18645c7d7b3546c73658dfd560b9a5a27.